org.eclipse.ui.views.pdf/src/org/eclipse/ui/views/pdf/PdfViewPage.java

```diff
@@ -320,89 +343,115 @@ private void refreshToolbar() {
 	/**
 	 * The hyperlink annotations in the PDF file per page.
 	 */
-	private final List<List<PdfAnnotation>> annotations = new ArrayList<List<PdfAnnotation>>();
+	private final Map<Integer, List<PdfAnnotation>> annotations = new HashMap<Integer, List<PdfAnnotation>>();
+	private Integer pageWithPriorityToLoad=null;
```

Comment:
Afterwards, I will also extract some methods in `loadAnnotationsJob` to improve readability.